src/wiring_diagrams/Undirected.jl

```diff
@@ -68,38 +68,45 @@ UndirectedWiringDiagram(port_types::AbstractVector{T}) where T =
 # Imperative interface
 ######################
 
-outer_box(::AbstractUWD) = 0
-box(d::AbstractUWD, args...) = subpart(d, args..., :box)
-junction(d::AbstractUWD, args...; outer::Bool=false) =
+# Accessors
+#----------
+
+outer_box(::AbstractACSet) = 0
+box(d::AbstractACSet, args...) = subpart(d, args..., :box)
+junction(d::AbstractACSet, args...; outer::Bool=false) =
   subpart(d, args..., outer ? :outer_junction : :junction)
 
-function junction(d::AbstractUWD, port::Tuple{Int,Int})
+function junction(d::AbstractACSet, port::Tuple{Int,Int})
   box, nport = port
   box == outer_box(d) ?
     junction(d, nport, outer=true) : junction(d, ports(d, box)[nport])
 end
 
-nboxes(d::AbstractUWD) = nparts(d, :Box)
-njunctions(d::AbstractUWD) = nparts(d, :Junction)
-boxes(d::AbstractUWD) = parts(d, :Box)
-junctions(d::AbstractUWD) = parts(d, :Junction)
+nboxes(d::AbstractACSet) = nparts(d, :Box)
+njunctions(d::AbstractACSet) = nparts(d, :Junction)
+boxes(d::AbstractACSet) = parts(d, :Box)
```

Comment:
Not necessary for this PR, but do we want to generate methods for every Object and Hom for the ACSets just have a convention that if you have a hom called f, then you get a julia function called `f(acs::ACSettype) = subpart(acs, :f)` and the same for objects nXs? Maybe that will lead to too many methods?

Reply:
Good question. I go back and forth on it. On one hand, writing these accessors leads to boilerplate; on the other hand, we'd have to do more code gen, which is a downside in its own right, and it's also often useful to tweak the API in ways that couldn't be automatically code gen-ed.